### PR TITLE
Zomars/cal 884 paid events not sending the link

### LINKED ISSUE #6663: [CAL-884] paid events not sending the link
For paid events, whenever an attendee completes the payment they don’t get the Google Meet link/Cal Video link in the invite. The just mentions Google or Cal Video as the location for the event but there is no link

<sub>[CAL-884](https://linear.app/calcom/issue/CAL-884/paid-events-not-sending-the-link)</sub>

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

- fixes #6663
- Extracts confirmation logic to be reused into handleConfirmation
- Re-uses logic in stripe webhook when a booking is paid and doesn't require confirmation

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Create a paid booking that doesn't require confirmation
- [ ] The Meeting URL should be in the email

### CODE REVIEW

File: packages/features/ee/payments/api/webhook.ts [hunk 1]
```diff
@@ -63,22 +80,14 @@ async function handlePaymentSuccess(event: Stripe.Event) {
     },
   });
 
+  console.log("booking", JSON.stringify(booking));
```
Comment (emrysal) [inline packages/features/ee/payments/api/webhook.ts:83]: Wanna keep this?

**Review by emrysal — APPROVED** [review]

File: packages/app-store/stripepayment/lib/PaymentService.ts [hunk 2]
```diff
@@ -89,7 +89,7 @@ export class PaymentService implements IAbstractPaymentService {
             stripe_publishable_key: this.credentials.stripe_publishable_key,
             stripeAccount: this.credentials.stripe_user_id,
           }) as unknown as Prisma.InputJsonValue,
-          fee: paymentFee,
+          fee: 0,
```
Comment (emrysal) [inline packages/app-store/stripepayment/lib/PaymentService.ts:92]: Assuming this is intentional BTW

File: packages/features/ee/payments/api/webhook.ts (hunk 1 above)
Comment (zomars) [reply packages/features/ee/payments/api/webhook.ts:83]: nope

File: packages/app-store/stripepayment/lib/PaymentService.ts (hunk 2 above)
Comment (zomars) [reply packages/app-store/stripepayment/lib/PaymentService.ts:92]: Will keep it out of this PR for now. But basically we might want to do this to support multiple countries.

File: packages/features/ee/payments/api/webhook.ts [hunk 3]
```diff
@@ -155,7 +162,11 @@ async function handlePaymentSuccess(event: Stripe.Event) {
 
   await prisma.$transaction([paymentUpdate, bookingUpdate]);
 
-  await sendScheduledEmails({ ...evt });
+  if (!isConfirmed && !eventTypeRaw?.requiresConfirmation) {
```
Comment (hariombalhara) [inline packages/features/ee/payments/api/webhook.ts:165]: Can we have a comment here as to what we are trying to do? They way I see it it's handling confirmation for a booking that doesn't need confirmation

File: packages/features/ee/payments/api/webhook.ts [hunk 4]
```diff
@@ -155,7 +162,11 @@ async function handlePaymentSuccess(event: Stripe.Event) {
...
+    await handleConfirmation({ user, evt, prisma, bookingId: booking.id, booking, paid: true });
+  } else {
+    await sendScheduledEmails({ ...evt });
```
Comment (hariombalhara) [inline packages/features/ee/payments/api/webhook.ts:168]: How is the case where isConfirmed is false and event requires confirmation? `sendScheduledEmails` seem to send confirmed booking emails only.

**Review by hariombalhara — COMMENTED** [review]
Left a comment

File: packages/features/ee/payments/api/webhook.ts (hunk 4 above)
Comment (zomars) [reply packages/features/ee/payments/api/webhook.ts:168]: If the booking is not already confirmed and doesn't require confirmation. I didn't want to modify too much the current logic

**Comment by zomars** [comment]
Merging due to client urgency. Agreed that this can be improved further. Will do in follow up.

**Pull request merged**